WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -588,26 +589,64 @@ private extension CardReaderConnectionController {
-                guard let adminUrl = url else {
-                    return
-                }
-                UIApplication.shared.open(adminUrl)
-                self?.showIncompleteAddressErrorWithRefreshButton()
-            }
             alerts.connectingFailedIncompleteAddress(from: from,
-                                                  adminUrl: adminUrl,
-                                                  site: ServiceLocator.stores.sessionManager.defaultSite,
-                                                  openUrlInSafari: openUrlInSafari,
-                                                  retrySearch: retrySearch,
-                                                  cancelSearch: cancelSearch)
+                                                     openWCSettings: openWCSettingsAction(adminUrl: adminUrl,
+                                                                                          from: from,
+                                                                                          retrySearch: retrySearch),
+                                                     retrySearch: retrySearch,
+                                                     cancelSearch: cancelSearch)
         case .invalidPostalCode:
             alerts.connectingFailedInvalidPostalCode(from: from, retrySearch: retrySearch, cancelSearch: cancelSearch)
         default:
             alerts.connectingFailed(from: from, continueSearch: continueSearch, cancelSearch: cancelSearch)
         }
     }
 
+    private func openWCSettingsAction(adminUrl: URL?,
+                                      from viewController: UIViewController,
+                                      retrySearch: @escaping () -> Void) -> ((UIViewController) -> Void)? {
+        if let adminUrl = adminUrl {
+            if let site = ServiceLocator.stores.sessionManager.defaultSite,
+               site.isWordPressStore {
+                return { [weak self] viewController in
+                    self?.openWCSettingsInWebview(url: adminUrl, from: viewController, retrySearch: retrySearch)
+                }
+            } else {
+                return { [weak self] _ in
+                    UIApplication.shared.open(adminUrl)
+                    self?.showIncompleteAddressErrorWithRefreshButton()
+                }
+            }
+        }
+        return nil
+    }
+    private func openWCSettingsInWebview(url adminUrl: URL,
+                                         from viewController: UIViewController,
+                                         retrySearch: @escaping () -> Void) {
+        let nav = NavigationView {
+            AuthenticatedWebView(isPresented: .constant(true),
+                                 url: adminUrl,
+                                 urlToTriggerExit: nil,
+                                 exitTrigger: nil)
+                                 .navigationTitle(CardPresentModalConnectingFailedUpdateAddress.Localization.adminWebviewTitle)
```

Comment:
I would have expected this string to remain in the CardReaderConnectionController.

Reply:
Yeah I wasn't sure about that, since it's much more related to the error message itself, but I've moved it over since I don't really want it to be confusing